### ISSUE TITLE
NullReferenceException in SymbolTable.GetParagraph

### ISSUE THREAD

**Issue by fm-117** [TypeCobolTeam/TypeCobol#2185]
**Describe the bug**
Exception in `SymbolTable.GetParagraph`, more precisely NullReferenceException is thrown in predicate defined here:

https://github.com/TypeCobolTeam/TypeCobol/blob/4b256f395eb8ac6a3a7fd3f70a52ba5d6d0483a9/TypeCobol/Compiler/CodeModel/SymbolTable.cs#L774

Full StackTrace is:
```
   à TypeCobol.Compiler.CodeModel.SymbolTable.<>c.<GetParagraph>b__37_2(Paragraph p)
   à System.Collections.Generic.List`1.FindAll(Predicate`1 match)
   à TypeCobol.Compiler.CodeModel.SymbolTable.GetParagraph(SymbolReference symbolRef, Section sectionNode)
   à TypeCobol.Compiler.CodeModel.SymbolTable.GetSectionOrParagraph(SymbolReference target, Section callerNodeSection)
   à TypeCobol.Compiler.Diagnostics.SectionOrParagraphUsageChecker.ResolveParagraphOrSection(Node callerNode, SymbolReference target, Section currentSection)
   à TypeCobol.Compiler.Diagnostics.CrossCompleteChecker.Visit(PerformProcedure performProcedureNode)
   à TypeCobol.Compiler.Nodes.PerformProcedure.VisitNode(IASTVisitor astVisitor)
   à TypeCobol.Compiler.Nodes.Node.AcceptASTVisitor(IASTVisitor astVisitor)
   à TypeCobol.Compiler.Nodes.Node.VisitChildrens(IASTVisitor astVisitor, Boolean continueVisit, IEnumerable`1 childrenNodes)
   à TypeCobol.Compiler.Nodes.Node.AcceptASTVisitor(IASTVisitor astVisitor)
   à TypeCobol.Compiler.Nodes.Node.VisitChildrens(IASTVisitor astVisitor, Boolean continueVisit, IEnumerable`1 childrenNodes)
   à TypeCobol.Compiler.Nodes.Node.AcceptASTVisitor(IASTVisitor astVisitor)
   à TypeCobol.Compiler.Nodes.Node.VisitChildrens(IASTVisitor astVisitor, Boolean continueVisit, IEnumerable`1 childrenNodes)
   à TypeCobol.Compiler.Nodes.Node.AcceptASTVisitor(IASTVisitor astVisitor)
   à TypeCobol.Compiler.Nodes.Node.VisitChildrens(IASTVisitor astVisitor, Boolean continueVisit, IEnumerable`1 childrenNodes)
   à TypeCobol.Compiler.Nodes.Node.AcceptASTVisitor(IASTVisitor astVisitor)
   à TypeCobol.Compiler.Parser.ProgramClassParserStep.CrossCheckPrograms(SourceFile root, TemporarySemanticDocument temporarySemanticDocument, TypeCobolOptions compilerOptions)
   à TypeCobol.Compiler.CompilationUnit.RefreshProgramClassDocumentSnapshot()
   à TypeCobol.LanguageServer.Workspace.<RefreshSyntaxTree>g__RefreshNodes|93_0(<>c__DisplayClass93_0& )
   à TypeCobol.LanguageServer.Workspace.RefreshSyntaxTree(FileCompiler fileCompiler, SyntaxTreeRefreshLevel refreshLevel)
   à TypeCobol.LanguageServer.Workspace.<>c__DisplayClass91_0.<TimerEvent>g__Refresh|0()
   à TypeCobol.LanguageServer.TypeCobolServerHost.MessageHandler(IMessageHandler messageHandler, TypeCobolServer typeCobolServer)
```

It means that either `Parent` or `CodeElement` property is null.

**To Reproduce**
TODO find a source file to reproduce.
The exception may be related to an edit

**Expected behavior**
No exception.

**Comment by fm-117** [comment]
As no source could be used to reproduce, we add diagotic code to help find the cause. See #2275.

**Comment by fm-117** [comment]
New traces show that a Paragraph node has been created with a Sentence node as parent.
This could happen in `ProgramClassBuilder.CheckStartSentenceLastStatement`.

https://github.com/TypeCobolTeam/TypeCobol/blob/fb301c4222ae7c2a41e4acbc4a36e8ad0f98da0b/TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilder.cs#L909-L926

Check why this hack is needed and either:
- find another workaround ?
- allow paragraph's parent code elements to be null in the GetParagraph method ?

**Comment by smedilol** [comment]
I see 3 solutions:
- Keep Sentence Node but fix the bug in `ProgramClassBuilder`. 
Method `StartParagraph` can `Exit` any current `Sentence` or `Paragraph` to be sure that the new `Paragraph` is correctly attached to a `Procedure division` or `Section`.
- Same but remove `Sentence` from `TypeCobolProgram.cup`. 
As `Sentence` is not a Cobol syntax it could be managed only in `ProgramClassBuilder`.
In this case `CheckStartSentenceLastStatement` can also be removed from `TypeCobolProgram.cup`. 
- Remove `Sentence` node completely.


> allow paragraph's parent code elements to be null in the GetParagraph method ?

This makes no sence from a Cobol perspective as a paragraph always has a parent.
Unless we consider to parse only a copy of code.

**Comment by fm-117** [comment]
Closing this as it is likely fixed with #2434. We'll keep traces for invalid paragraph parent active for a while so as to confirm the bug is fully fixed.
